File: Week 1/C programming/prog4.c
#include <stdio.h>

int main() {
    int arr[100], n, count = 0;

    printf("Enter number of elements: ");
    scanf("%d", &n);

    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Move all non-zero elements to front
    for (int i = 0; i < n; i++) {
        if (arr[i] != 0) {
            arr[count++] = arr[i];
        }
    }

    // Fill remaining positions with 0
    while (count < n) {
        arr[count++] = 0;
    }

    printf("Array after moving zeros to end:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}

File: Week 1/Python/prog4.py
arr = [0, 3, 0, 4, 5, 0, 1]

count = 0  # Count of non-zero elements

for i in range(len(arr)):
    if arr[i] != 0:
        arr[count] = arr[i]
        count += 1

# Fill remaining positions with 0
while count < len(arr):
    arr[count] = 0
    count += 1

print(arr)
